A30-xds-v3.md

```diff
@@ -0,0 +1,275 @@
+A30: xDS v3 Support
+----
+* Author(s): Mark D. Roth (markdroth)
+* Approver: ejona86, dfawley
+* Status: In Review
+* Implemented in: 
+* Last updated: 2020-06-25
+* Discussion at: https://groups.google.com/g/grpc-io/c/YRVqRMUQwhM
+
+## Abstract
+
+The purpose of this proposal is to describe how gRPC will handle the
+transition from v2 to v3 of the xDS API and to document how gRPC will handle
+xDS versioning on an ongoing basis.
+
+## Background
+
+gRPC has recently added support for obtaining configuration via the
+[xDS API](https://github.com/envoyproxy/data-plane-api).  Our current
+implementation uses v2 of the xDS API, but we are now adding support
+for [xDS
+v3](https://docs.google.com/document/d/1xeVvJ6KjFBkNjVspPbY_PwEDHC7XPi0J5p1SqUXcCl8/edit).
+In addition, we have been working with the xDS community on the new
+[minor/patch versioning
+proposal](https://docs.google.com/document/d/1afQ9wthJxofwOMCAle8qF-ckEfNWUuWtxbmdhxji0sI/edit),
+which should provide a smoother transition mechanism going forward.
+
+Regardless of whether a new version is indicated by bumping the
+major version (as was done with the v2 to v3 transition) or by bumping
+the minor version (as will probably be done for most new versions going
+forward), the current plan is that there will still be one new version of
+the xDS API per year.
+
+### Related Proposals: 
+
+[A27: xDS-Based Global Load Balancing](https://github.com/grpc/proposal/blob/master/A27-xds-global-load-balancing.md)
+
+## Proposal
+
+### Version Support Policy
+
+gRPC's policy will be to support two versions of xDS at any given time.
```

Comment:
Is this "at most two versions"?  Or "exactly two versions"?  If v4 isn't released for 5 years, will we support v2 for that long?

Reply:
The current plan is to have one new version per year, so I think this distinction isn't important.  If the version release schedule changes, then we can change our policy at that point.

Reply:
That sounds fair.  It does seem likely there will be an xDS version bump every time there is an opportunity to do so.